MDANSE/Src/MDANSE/Framework/Jobs/TrajectoryFilter.py

```diff
@@ -0,0 +1,279 @@
+            "instance"
+        ].chemical_system.atom_list
+
+        self._selected_atoms = [
+            self._atoms[i]
+            for i in np.array(self.configuration["atom_selection"]["indices"]).flatten()
+        ]
+
+        # This stores the trajectory (position array) of atoms by x, y, z component, to be filtered
+        self.atomic_trajectory_array = np.zeros(
+            (len(self._selected_atoms), 3, len(self.configuration["frames"]["value"]))
+        )
+
+    def run_step(self, index):
+        """
+        Runs a single step of the job.
+
+        :Parameters:
+            #. index (int): The index of the step.
+        """
+        LOG.debug(f"Running step: {index}")
+        trajectory = self.configuration["trajectory"]["instance"]
+
+        # get atom index
+        indexes = self.configuration["atom_selection"]["indices"][index]
+
+        series = trajectory.read_com_trajectory(
+            indexes,
+            first=self.configuration["frames"]["first"],
+            last=self.configuration["frames"]["last"] + 1,
+            step=self.configuration["frames"]["step"],
+        )
+
+        self.atomic_trajectory_array[index] = series.T
+
+        return index, None
+
+    def combine(self, index, x):
+        """
+        Combines returned results of run_step.
+        :Parameters:
+            #. index (int): The index of the step.
+            #. x (any): The returned result(s) of run_step
+        """
+        pass
+
+    def finalize(self):
+        """
+        Finalizes the calculations (e.g. averaging the total term, output files creations ...).
+        """
+
+        # Get filter class and instantiate filter object
+        filter_config = json.loads(self.configuration["trajectory_filter"]["value"])
+
+        filter_class, filter_attributes = (
+            FILTER_MAP[filter_config["filter"]],
+            filter_config["attributes"],
+        )
+
+        if {"n_steps", "time_step_ps"} not in set(filter_attributes.keys()):
+            filter_attributes.update(
+                {
+                    "n_steps": self.configuration["trajectory"]["length"],
+                    "time_step_ps": self.configuration["trajectory"]["md_time_step"],
+                }
+            )
+
+        filter = filter_class(**filter_attributes)
+
+        trajectories = copy.deepcopy(self.atomic_trajectory_array)
+
+        # Magnitude of zero frequency in filter response (equivalent to the average atomic positions)
+        zero_magnitude = np.abs(filter.freq_response.magnitudes[0])
+
+        # Apply filter (only apply initial position offset to atoms if filter response f(0) != 1)
+        filtered_coords = apply(
+            filter,
+            trajectories,
+            apply_offsets=not np.isclose(zero_magnitude, 1),
+        )
+
+        # Create new chemical system for output trajectory
+        name = self.configuration["output_files"]["file"].name.split(".").pop(0)
```

Comment:
Are you trying to get the basename?
```suggestion
        name = self.configuration["output_files"]["file"].stem
```